proposals/bip-002.md

```diff
@@ -0,0 +1,93 @@
+# Explicit Annotated status and completeness information
+
+| Field | Value |
+| --- | --- |
+| BIP | 2 |
+| Title | Explicit Annotated status and completeness information |
+| Author | Juanjo Álvarez |
+| Status | Draft |
+| Created | 2017-06-01 |
+| Updated | 2017-06-01 |
+
+## Abstract
+
+This BIP proposed a new official (as in defined in the Node struct and
+documented) "AnnotationStatus" property that would mark if the role has been
+annotated in any way and the quality of the current annotation.
+
+## Rationale
+
+At the current stage of the Babelfish project, we strive for genericity in the list
+of UAST roles and not adding very language-specific features into it until we can
+have a wider view about how to integrate or compose these roles for covering as
+much ground as possible. 
+
+This is a worthy objective but has the problem of leaving the nodes related with
+these features unannotated or with annotations that don't fully express all
+the semantic information that the language source code or the native AST have.
+
+This BIP proposes to add a new property in the Nodes called "AnnotationStatus"
+with the possible values of "Annotated", "Unannotated" and "Incomplete". This will
+make the UAST user have a wider view of the completeness of the information
+provided by it on specific nodes and also could help debug drivers and provide a
+completeness summary, with the added benefit of providing a numerically
+constatable measure of the completeness of every developed driver.
```

Comment:
The % coverage would indicate how many nodes are annotated which is something. They could be wrong because of bugs in the annotation code, but checking that is outside the scope of this BIP and yes, we definitely need to add UAST version checking eventually (that has been already talked about previously). Anyway I'll add a comment about this in the BIP.